### PR TITLE
Alec/tfac 377

### PR DESCRIPTION
**Pull request by amaliwanag2**

### CODE REVIEW

File: src/pages/beta-opt-in.js [hunk 1]
```diff
@@ -6,47 +6,44 @@ import Link from 'src/components/Link'
...
-      headers: new Headers({
-        // This custom header provides modest CSRF protection. See:
-        // https://github.com/gladly-team/tab-web#authentication-approach
-        'X-Gladly-Requested-By': 'tab-web-nextjs',
-        'Content-Type': 'application/json',
-      }),
-      credentials: 'include',
-      body: JSON.stringify({ optIn: isOptedIn }),
-    })
+export const setOptIn = async (isOptedIn) => {
```
Comment (amaliwanag2) [inline src/pages/beta-opt-in.js:9]: all i did was pull this out and export it

**Comment by amaliwanag2** [comment]
@kmjennison you can only test this on dev really, as the reload won't redirect you.  I went ahead and merged this onto dev and tested

File: src/__tests__/pages/index.test.js [hunk 2]
```diff
@@ -409,6 +411,55 @@ describe('index.js', () => {
...
+    expect(setOptIn).toHaveBeenCalledWith(false)
+    await act(async () => {
+      await flushAllPromises()
+      wrapper.update()
+    })
+    expect(reload).toHaveBeenCalled()
+    expect(setOptIn).toHaveBeenCalledTimes(1)
+  })
+
+  it('does not redirect to legacy if v4BetaEnabled is set to false', async () => {
```
Comment (kmjennison) [inline src/__tests__/pages/index.test.js:439]: Fix: "set to true"

File: src/pages/index.js [hunk 3]
```diff
@@ -331,6 +334,18 @@ const Index = ({ data: initialData }) => {
     }
   }, [globalTabCount])
 
+  // set v4 beta cookie to false if meant to be on legacy
+  const v4BetaEnabled = usePrevious(user ? user.v4BetaEnabled : undefined)
```
Comment (kmjennison) [inline src/pages/index.js:338]: @amaliwanag2 Why the `usePrevious`?

**Review by kmjennison — CHANGES_REQUESTED** [review]

File: src/pages/index.js (hunk 3 above)
Comment (amaliwanag2) [reply src/pages/index.js:338]: this was sometimes firing multiple times so I just did this defensively to only capture when it goes from undefined to true or false

**Review by kmjennison — APPROVED** [review]
LGTM 🍝

**Pull request merged**